api/app/schemas/hfi_calc.py

```diff
@@ -41,6 +42,101 @@ class StationDailyResponse(BaseModel):
+    grass_cure_percentage: Optional[float] = None
+    precipitation: Optional[float] = None
+    ffmc: Optional[float] = None
+    dmc: Optional[float] = None
+    dc: Optional[float] = None
+    isi: Optional[float] = None
+    bui: Optional[float] = None
+    fwi: Optional[float] = None
+    danger_class: Optional[int] = None
+    observation_valid: Optional[bool] = None
+    observation_valid_comment: Optional[str] = None
+    rate_of_spread: Optional[float] = None
+    hfi: Optional[float] = None
+    intensity_group: Optional[int] = None
+    sixty_minute_fire_size: Optional[float] = None
+    fire_type: Optional[str] = None
+    error: Optional[bool] = False
+    error_message: Optional[str] = None
+    date: Optional[datetime] = None
+    valid: bool
+
+
+class FireStartRange(BaseModel):
+    """
+    User facing label, value and lookup table of fire starts to prep level
+    """
+    label: str
+    value: int
+    lookup_table: Mapping[int, int]
+
+
+lowest_fire_starts = FireStartRange(label='0-1', value=1, lookup_table={1: 1, 2: 1, 3: 2, 4: 3, 5: 4})
+one_2_two_starts = FireStartRange(label='1-2', value=2, lookup_table={1: 1, 2: 2, 3: 3, 4: 4, 5: 5})
+two_2_three_starts = FireStartRange(label='2-3', value=3, lookup_table={1: 2, 2: 3, 3: 4, 4: 5, 5: 6})
+three_2_six_starts = FireStartRange(label='3-6', value=6, lookup_table={1: 3, 2: 4, 3: 5, 4: 6, 5: 6})
+highest_fire_starts = FireStartRange(label='6+', value=7, lookup_table={1: 4, 2: 5, 3: 6, 4: 6, 5: 6})
+
+
+class DailyResult(BaseModel):
+    """
+    Prep level, MIG, fire starts and station daily results for a day in a prep week
+    """
+    dateISO: str
+    dailies: List[ValidatedStationDaily]
+    fire_starts: FireStartRange
+    mean_intensity_group: Optional[float]
+    prep_level: Optional[float]
+
+
+class PlanningAreaResult(BaseModel):
+    """
+    Mean prep level / max intensity group,
+    dailies and validity status of station dailies in a planning area
+    """
+    all_dailies_valid: bool
+    highest_daily_intensity_group: Optional[float]
+    mean_prep_level: Optional[float]
+    daily_results: List[DailyResult]
+
+
+class HFIResultResponse(BaseModel):
+    """
+    Response that contains daily data, num prep days, selected station codes,
+    selected fire centre, fire starts, HFI results
+    """
+    num_prep_days: int
+    selected_prep_date: datetime
+    start_time_stamp: Optional[int]
+    end_time_stamp: Optional[int]
+    selected_station_codes: List[int]
+    selected_fire_center: Optional[str]
+    planning_area_hfi_results: Mapping[str, PlanningAreaResult]
+    planning_area_fire_starts: Mapping[str, List[FireStartRange]]
+
+
+class HFIResultRequest(BaseModel):
+    num_prep_days: int
+    selected_prep_date: datetime
+    start_time_stamp: Optional[int]
+    end_time_stamp: Optional[int]
+    selected_station_codes: List[int]
+    selected_fire_center: Optional[str]
+    planning_area_fire_starts: Mapping[str, List[FireStartRange]]
```

Comment:
Similar comment on planning area key - I'm guessing that key should be an id - in which case it's an integer.

Reply:
Currently, the frontend doesn't know about the planning area id's it just knows it's name: https://github.com/bcgov/wps/blob/d3b180a9246a34404a4ad8eb6148be9d27e9f1b1/api/app/schemas/hfi_calc.py#L58

Reply:
It would save me time if you did, but you could argue it's out of scope for your PR. You didn't have that re-factor as part of your scope, and it isn't required for your story. Up to you if you want to do it now.

Reply:
Added to: https://github.com/bcgov/wps/pull/1705#issuecomment-1035265778